### PR TITLE
CI: Expose detected plugin name to other workflow jobs

### PR DESCRIPTION
**Pull request by PatTheMav**
### Description
Adds early detection of the configured plugin name in workflows so the name can be re-used in later workflows without the need to do a full repository checkout or installation of 3rd party utilities (e.g. the release creation step).

### Motivation and Context
Reduce necessary fixup steps for plugin developers when finalising their releases.

Fixes https://github.com/obsproject/obs-plugintemplate/issues/105.

### How Has This Been Tested?
Tested with a tagged release on a fork of the repository.

### Types of changes
- Performance enhancement (non-breaking change which improves efficiency)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: .github/workflows/build-project.yaml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 name: Build Project
 on:
   workflow_call:
+    outputs:
+      pluginName:
+        description: 'Project name detected by parsing build spec file'
+        value: ${{ jobs.check-event.outputs.pluginName}}
```
Comment (RytoEX) [inline .github/workflows/build-project.yaml:7]:
```suggestion
        value: ${{ jobs.check-event.outputs.pluginName }}
```

**Review by RytoEX — CHANGES_REQUESTED** [review]

**Review by RytoEX — APPROVED** [review]
Seems fine.

**Comment by CodeYan01** [comment]
With the new `displayName` change, have you tested creating a new release, with a space in the displayName? I am getting an error: https://github.com/CodeYan01/media-playlist-source/actions/runs/7257209427/job/19770748632

**Comment by PatTheMav** [comment]
> With the new `displayName` change, have you tested creating a new release, with a space in the displayName? I am getting an error: https://github.com/CodeYan01/media-playlist-source/actions/runs/7257209427/job/19770748632

Right, it's not supposed to be used for file names now that it can have spaces and other weird characters in it, forgot to remove that again.

**Comment by PatTheMav** [comment]
@CodeYan01 pls check again.

**Comment by CodeYan01** [comment]
In the Issue, you commented that it will fallback to `name` if `displayName` is not available, but I'm getting an error when the field is not there. https://github.com/CodeYan01/media-playlist-source/actions/runs/7284773565

With the `displayName` it now works perfectly. https://github.com/CodeYan01/media-playlist-source/actions/runs/7284781779

Thank you!

**Comment by PatTheMav** [comment]
> In the Issue, you commented that it will fallback to `name` if `displayName` is not available, but I'm getting an error when the field is not there. https://github.com/CodeYan01/media-playlist-source/actions/runs/7284773565
> 
> With the `displayName` it now works perfectly. https://github.com/CodeYan01/media-playlist-source/actions/runs/7284781779
> 
> Thank you!

Ah, good old `pipefail`, should be fixed in a bit.

**Comment by CodeYan01** [comment]
Tested both cases, works now.

**Pull request merged**